src/components/AssetLogo.tsx

```diff
@@ -1,26 +1,47 @@
 import Image from "next/image"
-import { FC } from "react"
+import { CSSProperties, FC } from "react"
 
+import useCurrencyLogoURI from "@/hooks/useCurrencyLogoURIs"
+
+import AuraLogo from "~/images/assets/aura.png"
 import BalancerLogo from "~/images/assets/balancer.png"
 import CurveLogo from "~/images/assets/curve.png"
 
 type AssetLogoProps = {
   name: string
   className: string
+  style?: CSSProperties
+  tokenAddress?: `0x${string}`
 }
 
-const AssetLogo: FC<AssetLogoProps> = ({ className, name }) => {
-  switch (name) {
-    case "curve":
-      return <Image src={CurveLogo} className={className} priority alt="" />
-    case "balancer":
-      return <Image src={BalancerLogo} className={className} priority alt="" />
-    case "token":
-      // use CurveLogo for now
-      return <Image src={CurveLogo} className={className} priority alt="" />
-    default:
-      return null
+const AssetLogo: FC<AssetLogoProps> = ({
+  className,
+  name,
+  tokenAddress,
+  style,
+}) => {
+  const { logoURI } = useCurrencyLogoURI(tokenAddress || "0x")
+
+  const getImage = (name: string) => {
+    switch (name) {
+      case "curve":
+        return <Image src={CurveLogo} priority alt="" />
+      case "balancer":
+        return <Image src={BalancerLogo} priority alt="" />
+      case "crypto":
```

Comment:
We have only aurabal in crypto now but it won't be the case in the future. It's not a better way to handle the crypto category even if it's working currently imo.